src/NLog/Targets/ConsoleWordHighlightingRule.cs

```diff
@@ -157,9 +195,22 @@ private string MatchEvaluator(Match m)
             return result.ToString();
         }
 
+
         internal string ReplaceWithEscapeSequences(string message)
         {
-            return this.CompiledRegex.Replace(message, new MatchEvaluator(this.MatchEvaluator));
+            if (CompileRegex)
+            {
+                return this.CompiledRegex.Replace(message, this.MatchEvaluator);
```

Comment:
Perhaps we should check if `CompiledRegex` is null here and then return just `message`.